File: src/main/java/com/github/nigelzor/hogs/TileFactory.java
package com.github.nigelzor.hogs;

import java.util.EnumSet;

public class TileFactory {

	/**
	 * west-east
	 */
	public static Tile straight() {
		return new Tile(EnumSet.of(Direction.WEST, Direction.EAST));
	}

	/**
	 * west-south
	 */
	public static Tile elbow() {
		return new Tile(EnumSet.of(Direction.WEST, Direction.SOUTH));
	}

	/**
	 * west-north-east
	 */
	public static Tile tee() {
		return new Tile(EnumSet.of(Direction.WEST, Direction.SOUTH, Direction.EAST));
	}

	/**
	 * west-east, objective one
	 */
	public static Tile homework() {
		return new Tile(EnumSet.of(Direction.WEST, Direction.EAST), Objective.ONE);
	}

	/**
	 * north-south, objective two
	 */
	public static Tile potions() {
		return new Tile(EnumSet.of(Direction.NORTH, Direction.SOUTH), Objective.TWO);
	}

	/**
	 * north-east, objective three
	 */
	public static Tile creatures() {
		return new Tile(EnumSet.of(Direction.NORTH, Direction.EAST), Objective.THREE);
	}

	/**
	 * south, objective four
	 */
	public static Tile tower() {
		return new Tile(EnumSet.of(Direction.SOUTH), Objective.FOUR);
	}

}

File: src/main/kotlin/com/github/nigelzor/hogs/TileFactory.kt
package com.github.nigelzor.hogs

import kotlin.Int as BTile

object TileFactory {
	fun straight() = BTiles.fromDirections(Direction.WEST, Direction.EAST)
	fun elbow() = BTiles.fromDirections(Direction.WEST, Direction.SOUTH)
	fun tee() = BTiles.fromDirections(Direction.WEST, Direction.SOUTH, Direction.EAST)
	fun homework() = BTiles.fromDirections(Direction.WEST, Direction.EAST).with(Objective.ONE)
	fun potions() = BTiles.fromDirections(Direction.NORTH, Direction.SOUTH).with(Objective.TWO)
	fun creatures() = BTiles.fromDirections(Direction.NORTH, Direction.EAST).with(Objective.THREE)
	fun tower() = BTiles.fromDirections(Direction.SOUTH).with(Objective.FOUR)
}
